core/src/main/java/com/oath/oak/Chunk.java

```diff
@@ -42,7 +42,7 @@
     /*-------------- Members --------------*/
 
     private static final Unsafe unsafe = UnsafeUtils.unsafe;
-    ByteBuffer minKey;       // minimal key that can be put in this chunk
+    MinKeyReadBuffer minKey;       // minimal key that can be put in this chunk
```

Comment:
I understand the reason for some special case of buffer here. But the solution looks to me too narrow. At least I would create an "OnHeapOakAttachedReadBuffer" class. However, the best solution looks to me, to create just KeyBuffer or OakAttachedReadBuffer out of min key (copy it off-heap). Later in rebalance just reference to the relevant off-heap buffer for the min key, anyway in OakMap keys are never released (even if deleted) until chunk rebalances...

Reply:
Modify to allocate min-key using Oak's allocator.
Future work: to modify min-key to be inclusive, then we could use an existing object as min-key.